### PR TITLE
ISSUE-21: Implemented access token authentication to all tasks

### PR DESCRIPTION
**Pull request by Svenskapojkarna**
Implements #21

### CODE REVIEW

File: Frends.Community.Azure.Blob.Tests/DeleteTest.cs [hunk 1]
```diff
@@ -9,72 +9,122 @@ namespace Frends.Community.Azure.Blob.Tests
...
             var input = new DeleteBlobProperties
             {
                 BlobName = Guid.NewGuid().ToString()
             };
             var options = new BlobConnectionProperties
             {
+                ConnectionMethod = ConnectionMethod.ConnectionString,
                 ConnectionString = _connectionString,
-                ContainerName = Guid.NewGuid().ToString()
+                ContainerName = containerName
```
Comment (jefim) [inline Frends.Community.Azure.Blob.Tests/DeleteTest.cs:57]: Why is this changed to be some predefined value instead of a random Guid? Isn't the purpose of the test to check that non-existent container case is ok? What if "test2" container somehow stays alive after some tests?

File: Frends.Community.Azure.Blob.Tests/DeleteTest.cs [hunk 2]
```diff
@@ -9,72 +9,122 @@ namespace Frends.Community.Azure.Blob.Tests
...
+                TenantID = _tenantID,
+                ClientSecret = _clientSecret
+            };
+            var container = Utils.GetBlobContainer(_connectionString, containerName);
+            await container.CreateIfNotExistsAsync();
+
+            var result = await DeleteTask.DeleteBlobAsync(input, connection, new CancellationToken());
+
+            Assert.IsTrue(result.Success, "DeleteBlob should've returned true when trying to delete non existing blob");
+            await DeleteContainer(containerName);
```
Comment (jefim) [inline Frends.Community.Azure.Blob.Tests/DeleteTest.cs:90]: Is there any simple way to actually make sure that container was deleted or is this out of scope for this test? I mean for all we know the DeleteBlobAsync can just read "return true" and this test will pass :)

Comment (jefim) [reply Frends.Community.Azure.Blob.Tests/DeleteTest.cs:90]: I guess this comment above applies to some other unit tests as well.

File: Frends.Community.Azure.Blob/Definition.cs [hunk 3]
```diff
@@ -1,48 +1,10 @@
...
-        /// <summary>
-        /// How repeats of input are separated.
-        /// </summary>
-        [DisplayFormat(DataFormatString = "Text")]
-        [DefaultValue(" ")]
-        public string Delimiter;
-    }
-
-    public class Result
+    public enum ConnectionMethod
```
Comment (jefim) [inline Frends.Community.Azure.Blob/Definition.cs:5]: Since this is the only thing left in this file, then I think the file should be renamed to match the enum name.

File: Frends.Community.Azure.Blob/UploadTask.cs [hunk 4]
```diff
@@ -201,8 +206,16 @@ private static Encoding GetEncoding(string target)
             string fileName,
             CancellationToken cancellationToken)
         {
-            // get the destination blob, rename if necessary
-            var blob = new PageBlobClient(destinationProperties.ConnectionString, destinationProperties.ContainerName, fileName);
+            PageBlobClient blob;
+
+            if (destinationProperties.ConnectionMethod == ConnectionMethod.ConnectionString)
```
Comment (jefim) [inline Frends.Community.Azure.Blob/UploadTask.cs:211]: This comment addresses all of blob and container client creation (in all tasks really). I see that all tasks have different classes that contain the same properties for connection. Can we extract an interface like
```
public interface IStorageAccountConnection
{
    enum Method = ConnectionString / ClientCredentials;
    string ConnectionString = ...
    string Clientid, secret, whatever else = ...
```
And then in some Util class make two methods like
```
class Utils
{
    public BlobClient CreateBlobClient(IStorageAccountConnection conn)
    { ... }

    public ContainerClient CreateContainerClient(IStorageAccountConnection conn)
    { ... }
}
```
This would remove quite a large portion of copy-pasted connection code. In every task you could just `Utils.CreateBlobClient(source)`.

File: Frends.Community.Azure.Blob/Definition.cs [hunk 5]
```diff
@@ -1,48 +1,10 @@
...
     {
-        /// <summary>
-        /// Contains input repeated specified times.
-        /// </summary>
-        [DisplayFormat(DataFormatString = "Text")]
-        public string Replication;
+        ConnectionString,
+        AccessToken
```
Comment (jefim) [inline Frends.Community.Azure.Blob/Definition.cs:8]: Should this be called "Client Credentials" or "OAuth"? Because user does not provide the token, but instead provides the client credentials / utilizes OAuth for auth.

File: Frends.Community.Azure.Blob.Tests/DeleteTest.cs (hunk 2 above)
Comment (Svenskapojkarna) [reply Frends.Community.Azure.Blob.Tests/DeleteTest.cs:90]: That is not a test, it's a cleaner. There was an issue with cleanup where tests would randomly fail due to containers being deleted when they shouldn't be, so I decided to implement the cleaner as a helper method to which you can specify which container should be deleted.
